graph/migrate/src/lib/state/automatic/selectors.ts

```diff
@@ -4,58 +4,102 @@ import { MigrationDetailsWithId } from 'nx/src/config/misc-interfaces';
 /* eslint-enable @nx/enforce-module-boundaries */
 import type { AutomaticMigrationState } from './types';
 
-// this is where the ui should start off on
-// we assume completed migrations in the past are reviewed
 export function findFirstIncompleteMigration(
   migrations: MigrationDetailsWithId[],
   nxConsoleMetadata: MigrationsJsonMetadata
 ) {
   return (
-    migrations.find(
-      (migration) =>
-        nxConsoleMetadata.completedMigrations?.[migration.id]?.type !==
-          'successful' &&
-        nxConsoleMetadata.completedMigrations?.[migration.id]?.type !==
-          'skipped'
-    ) ?? migrations[0]
+    migrations.find((migration) => {
+      const type = nxConsoleMetadata.completedMigrations?.[migration.id]?.type;
+      return type !== 'successful' && type !== 'skipped';
+    }) ?? migrations[0]
   );
 }
 
 export function currentMigrationHasSucceeded(ctx: AutomaticMigrationState) {
   if (!ctx.currentMigration) {
     return false;
   }
-  const completedMigration =
-    ctx.nxConsoleMetadata?.completedMigrations?.[ctx.currentMigration.id];
-  return completedMigration?.type === 'successful';
+  return isMigrationSuccessful(ctx, ctx.currentMigration.id);
 }
 
 export function currentMigrationHasChanges(ctx: AutomaticMigrationState) {
   if (!ctx.currentMigration) {
     return false;
   }
-  const completedMigration =
-    ctx.nxConsoleMetadata?.completedMigrations?.[ctx.currentMigration.id];
+  const completed =
+    ctx.nxConsoleMetadata?.completedMigrations?.[ctx.currentMigration?.id];
   return (
-    completedMigration?.type === 'successful' &&
-    completedMigration.changedFiles.length > 0
+    completed?.type === 'successful' &&
+    (completed.changedFiles?.length ?? 0) > 0
+  );
+}
+
+export function isMigrationSuccessful(
+  ctx: AutomaticMigrationState,
+  migrationId: string
+) {
+  return (
+    ctx.nxConsoleMetadata?.completedMigrations?.[migrationId]?.type ===
+    'successful'
+  );
+}
+
+export function isMigrationFailed(
```

Comment:
Instead of having one selector for each type, I'd just use `getMigrationType` and then in the component you can select the type and do the `===` comparison there. Otherwise we'd have to add more and more of these selectors if we add more types.